Tests/NIOHTTP2Tests/ConfiguringPipelineAsyncMultiplexerTests.swift

```diff
@@ -132,6 +132,91 @@ final class ConfiguringPipelineAsyncMultiplexerTests: XCTestCase {
 
         serverRecorder.receivedFrames.assertFramePayloadsMatch(Array(repeating: ConfiguringPipelineAsyncMultiplexerTests.requestFramePayload, count: requestCount))
     }
+
+    func testNIOAsyncChannelPipelineCommunicates() async throws {
```

Comment:
Can we have a test for the API which doesn't configure the connection channel as well?